support/mender-inventory-provides

```diff
@@ -10,6 +10,31 @@ set -e
 # Also submit device_type, which is in a separate file, not in the database. We don't know if the
 # file contains a newline, so it's important that this is done last, to avoid corrupting other
 # entries.
-grep '^device_type=' /var/lib/mender/device_type
+
+# Set default values for configuration directories
+mender_conf_dir=${MENDER_CONF_DIR:-/etc/mender}
+mender_datastore_dir=${MENDER_DATASTORE_DIR:-/var/lib/mender}
+
+# Set default value for DeviceTypeFile
+default_device_type_file="$mender_datastore_dir/device_type"
+
+# Poor man's case insensitive match.
+# Read mender.conf file to find the DeviceTypeFile key if it exists prioritizing /etc over /var/lib
+match="[Dd][Ee][Vv][Ii][Cc][Ee][Tt][Yy][Pp][Ee][Ff][Ii][Ll][Ee]"
+if [ -f "$mender_conf_dir/mender.conf" ]; then
+    device_type_file=$(sed -ne '/"'"$match"'" *: *"[^"]*"/ { s/.*"'"$match"'" *: *"\([^"]*\)".*/\1/; p }' "$mender_conf_dir/mender.conf" || true)
+fi
+
+if [ -z "$device_type_file" ] && [ -f "$mender_datastore_dir/mender.conf" ]; then
```

Comment:
If `$device_type_file` variable is defined (iow got a value from /etc/mender/mender.conf) but the device_type file does not exist, we'll not get here for the fallback case where a different file could be specified and could be valid.

I don't know how important this is to handle, though. Is just an observation while experimenting with your fix on my device.

Edit: to make clear my comment - it could be handled with something like:
```
if [ -f "$mender_datastore_dir/mender.conf" ]; then
    if [ -z "$device_type_file" ] || [ ! -f "$device_type_file" ]; then
        ...
    fi
fi
```